### PR TITLE
Improve GitHub action, add badge to README

### PR DESCRIPTION
**Pull request by DmytroLitvinov**
@sobolevn , improve GitHub action to also run CI on push to `master`. It will allow us to add badge for `master` branch in future.

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
-name: CI
+name: linter
```
Comment (sobolevn) [inline .github/workflows/linter.yml:1]: Let's keep the name unedited.

Because, we will have empty sections in the Actions dashboard otherwise.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks! 

Also see https://github.com/typeddjango/awesome-python-typing/pull/44#issuecomment-702379835

**Comment by DmytroLitvinov** [comment]
@sobolevn, done :+1:

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Awesome Python Typing [![Awesome](https://awesome.re/badge-flat2.svg)](https://awesome.re) [![Gitter](https://img.shields.io/gitter/room/mypy-django/Lobby?color=9cf&style=flat-square)][TypedDjango]
+# Awesome Python Typing [![Awesome](https://awesome.re/badge-flat2.svg)](https://awesome.re) [![Gitter](https://img.shields.io/gitter/room/mypy-django/Lobby?color=9cf&style=flat-square)][TypedDjango] [![CI](https://github.com/typeddjango/awesome-python-typing/workflows/CI/badge.svg?branch=master&event=push)](https://github.com/typeddjango/awesome-python-typing/actions?query=workflow%3ACI)
```
Comment (sobolevn) [inline README.md:1]: We store our build badge in `contributing.md`

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks!

**Pull request merged**